test/framework/src/main/java/org/elasticsearch/transport/nio/channel/TcpChannelFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.transport.nio.channel;
+
+
+import org.elasticsearch.transport.TcpTransport;
+import org.elasticsearch.transport.nio.AcceptingSelector;
+import org.elasticsearch.transport.nio.SocketSelector;
+
+import java.io.IOException;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.util.function.Consumer;
+
+public class TcpChannelFactory extends ChannelFactory<TcpNioServerSocketChannel, TcpNioSocketChannel> {
```

Comment:
can this be pkg private? and can we have javadocs?

Reply:
It cannot be package private as it is instantiated in a `NioTransport` which is in a different package. I think at some point I will rework the packages and it can eventually be package private. But that is a different PR.